internal/jsnum/pseudobigint.go

```diff
@@ -25,3 +31,37 @@ func (value PseudoBigInt) Sign() int {
 	}
 	return 1
 }
+
+func ParseValidBigInt(text string) PseudoBigInt {
+	text, negative := strings.CutPrefix(text, "-")
+	return PseudoBigInt{
+		Negative:    negative,
+		Base10Value: ParsePseudoBigInt(text),
+	}
+}
+
+func ParsePseudoBigInt(stringValue string) string {
+	stringValue = strings.TrimSuffix(stringValue, "n")
+
+	var b1 byte
```

Comment:
minor, but: having names `b1 bi big` in the same small space is way too hard to read.

Edit: good chance to promote https://www.codingfont.com/